### PR TITLE
[MNT] Update `pre-commit` `pyupgrade` config

### PR DESCRIPTION
**Pull request by MatthewMiddlehurst**
#### What does this implement/fix? Explain your changes.

Updates the `pyupgrade` configuration to drop Python 3.8, which we no longer support.

### CODE REVIEW

**Comment by MatthewMiddlehurst** [comment]
ofcourse

File: aeon/classification/distance_based/_proximity_forest.py [hunk 1]
```diff
@@ -89,7 +89,7 @@ def __init__(
         n_splitters: int = 5,
         max_depth: int = None,
         min_samples_split: int = 2,
-        random_state: Union[int, Type[np.random.RandomState], None] = None,
+        random_state: Union[int, type[np.random.RandomState], None] = None,
```
Comment (CodeLionX) [inline aeon/classification/distance_based/_proximity_forest.py:92]: Should be anyway:
```suggestion
        random_state: Union[int, np.random.RandomState, None] = None,
```

File: aeon/classification/distance_based/_proximity_tree.py [hunk 2]
```diff
@@ -120,7 +120,7 @@ def __init__(
         n_splitters: int = 5,
         max_depth: int = None,
         min_samples_split: int = 2,
-        random_state: Union[int, Type[np.random.RandomState], None] = None,
+        random_state: Union[int, type[np.random.RandomState], None] = None,
```
Comment (CodeLionX) [inline aeon/classification/distance_based/_proximity_tree.py:123]: Should be anyway:
```suggestion
        random_state: Union[int, np.random.RandomState, None] = None,
```

File: aeon/classification/shapelet_based/_rdst.py [hunk 3]
```diff
@@ -152,7 +152,7 @@ def __init__(
         save_transformed_data: bool = False,
         class_weight=None,
         n_jobs: int = 1,
-        random_state: Union[int, Type[np.random.RandomState], None] = None,
+        random_state: Union[int, type[np.random.RandomState], None] = None,
```
Comment (CodeLionX) [inline aeon/classification/shapelet_based/_rdst.py:155]: Should be anyway:
```suggestion
        random_state: Union[int, np.random.RandomState, None] = None,
```

File: aeon/classification/shapelet_based/_stc.py [hunk 4]
```diff
@@ -142,7 +142,7 @@ def __init__(
         contract_max_n_shapelet_samples: int = np.inf,
         n_jobs: int = 1,
         batch_size: Union[int, None] = 100,
-        random_state: Union[int, Type[np.random.RandomState], None] = None,
+        random_state: Union[int, type[np.random.RandomState], None] = None,
```
Comment (CodeLionX) [inline aeon/classification/shapelet_based/_stc.py:145]: Here as well:
```suggestion
        random_state: Union[int, np.random.RandomState, None] = None,
```

File: aeon/classification/sklearn/_continuous_interval_tree.py [hunk 5]
```diff
@@ -27,7 +27,7 @@ class _TreeNode:
     """ContinuousIntervalTree tree node."""
 
     def __init__(
-        self, random_state: Union[int, Type[np.random.RandomState], None] = None
+        self, random_state: Union[int, type[np.random.RandomState], None] = None
```
Comment (CodeLionX) [inline aeon/classification/sklearn/_continuous_interval_tree.py:30]: and in this file as well:
```suggestion
        self, random_state: Union[int, np.random.RandomState, None] = None,
```

File: aeon/classification/sklearn/_rotation_forest_classifier.py [hunk 6]
```diff
@@ -106,12 +106,12 @@ def __init__(
         min_group: int = 3,
         max_group: int = 3,
         remove_proportion: float = 0.5,
-        base_estimator: Optional[Type[BaseEstimator]] = None,
+        base_estimator: Optional[type[BaseEstimator]] = None,
```
Comment (CodeLionX) [inline aeon/classification/sklearn/_rotation_forest_classifier.py:109]:
```suggestion
        base_estimator: Optional[BaseEstimator] = None,
```
and in other locations in this file

File: aeon/regression/sklearn/_rotation_forest_regressor.py [hunk 7]
```diff
@@ -104,12 +104,12 @@ def __init__(
         min_group: int = 3,
         max_group: int = 3,
         remove_proportion: float = 0.5,
-        base_estimator: Optional[Type[BaseEstimator]] = None,
+        base_estimator: Optional[type[BaseEstimator]] = None,
```
Comment (CodeLionX) [inline aeon/regression/sklearn/_rotation_forest_regressor.py:107]:
```suggestion
        base_estimator: Optional[BaseEstimator] = None,
```
and other locations in this file

**Review by CodeLionX — DISMISSED** [review]

**Review by CodeLionX — APPROVED** [review]

**Pull request merged**